### PR TITLE
[1.11.0-GA] dcos-net not starting if IPv6 is disabled

### PR DESCRIPTION
**Pull request by urbanserj**
## High-level description

What features does this change enable? What bugs does this change fix?

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS-21471](https://jira.mesosphere.com/browse/DCOS-21471) dcos-net not starting if IPv6 is disabled

## Related tickets (optional)

Other tickets related to this change:

## Checklist for all PRs

  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [x] Test Results: [link to CI job test results for component]
  - [x] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by urbanserj** [comment]
@mesosphere-mergebot bump-ee

**Comment by urbanserj** [comment]
@mesosphere-mergebot label [Ready For Review]

File: gen/calc.py [hunk 1]
```diff
@@ -349,6 +349,23 @@ def validate_dcos_overlay_network(dcos_overlay_network):
...
+        overlay_network = json.loads(dcos_overlay_network)
+        del overlay_network['vtep_subnet6']
+        overlay_network['overlays'] = \
+            [o for o in overlay_network['overlays'] if 'subnet' in o]
+        return json.dumps(overlay_network)
+    else:
+        overlay_network = json.loads(dcos_overlay_network)
+        overlays = overlay_network['overlays']
+        assert [o['name'] for o in overlays if 'subnet6' in o] == [], \
+            "ipv6 is disabled, only ipv4 networks are allowed"
```
Comment (GoelDeepak) [inline gen/calc.py]: missing return in `else` case?

File: gen/calc.py [hunk 2]
```diff
@@ -515,6 +532,13 @@ def validate_dcos_l4lb_max_named_ip6(dcos_l4lb_max_named_ip6):
     validate_ipv6_addresses([dcos_l4lb_max_named_ip6])
 
 
+def validate_dcos_l4lb_enable_ipv6(dcos_l4lb_enable_ipv6, enable_ipv6):
+    validate_true_false(dcos_l4lb_enable_ipv6)
+    if enable_ipv6 == 'false':
+        assert dcos_l4lb_enable_ipv6 == 'false', "When enable_ipv6 is false, " \
+            "dcos_l4lb_enable_ipv6 mush be false as well"
```
Comment (GoelDeepak) [inline gen/calc.py]: s/mush/must

File: packages/dcos-net/extra/dcos-net.service [hunk 3]
```diff
@@ -24,10 +25,10 @@ ExecStartPre=/opt/mesosphere/active/dcos-net/dcos-net/bin/dcos-net-setup.py ip l
 ExecStartPre=/opt/mesosphere/active/dcos-net/dcos-net/bin/dcos-net-setup.py ip link set minuteman up
 ExecStartPre=/opt/mesosphere/active/dcos-net/dcos-net/bin/dcos-net-setup.py ip link add spartan type dummy
 ExecStartPre=/opt/mesosphere/active/dcos-net/dcos-net/bin/dcos-net-setup.py ip link set spartan up
-ExecStartPre=/opt/mesosphere/active/dcos-net/dcos-net/bin/dcos-net-setup.py sysctl -w net.ipv6.conf.spartan.disable_ipv6=0
 ExecStartPre=/opt/mesosphere/active/dcos-net/dcos-net/bin/dcos-net-setup.py ip addr add 198.51.100.1/32 dev spartan
 ExecStartPre=/opt/mesosphere/active/dcos-net/dcos-net/bin/dcos-net-setup.py ip addr add 198.51.100.2/32 dev spartan
 ExecStartPre=/opt/mesosphere/active/dcos-net/dcos-net/bin/dcos-net-setup.py ip addr add 198.51.100.3/32 dev spartan
-ExecStartPre=/opt/mesosphere/active/dcos-net/dcos-net/bin/dcos-net-setup.py ip -6 addr add fd01:d::c633:6401/128 dev spartan
+ExecStartPre=/opt/mesosphere/active/dcos-net/dcos-net/bin/dcos-net-setup.py --ipv6-only sysctl -w net.ipv6.conf.spartan.disable_ipv6=0
```
Comment (GoelDeepak) [inline packages/dcos-net/extra/dcos-net.service]: `--ipv6-only` is misleading. Could we just name the parameter as `--ipv6`?

File: packages/dcos-net/extra/dcos-net.service [hunk 4]
```diff
@@ -24,10 +25,10 @@ ExecStartPre=/opt/mesosphere/active/dcos-net/dcos-net/bin/dcos-net-setup.py ip l
...
+ExecStartPre=/opt/mesosphere/active/dcos-net/dcos-net/bin/dcos-net-setup.py --ipv6-only ip -6 addr add fd01:d::c633:6401/128 dev spartan
```
Comment (GoelDeepak) [inline packages/dcos-net/extra/dcos-net.service]: same comment as above

**Comment by asridharan** [comment]
@orsenthil just wondering how did this PR get a `ship it` label? Don't see any approvals on this as yet?

**Comment by orsenthil** [comment]
@asridharan  - That's because the status checks are not marked as "Required". I will turn them to required.

**Comment by urbanserj** [comment]
@mesosphere-mergebot bump-ee

**Comment by orsenthil** [comment]
@mesosphere-mergebot override-status teamcity/dcos/test/azure/arm https://jira.mesosphere.com/browse/QUALITY-1881

**Comment by urbanserj** [comment]
@mesosphere-mergebot bump-ee

File: gen/dcos-config.yaml [hunk 5]
```diff
@@ -384,8 +385,13 @@ package:
               "network_mode": "USER",
               "dns": {
                 "nameservers": [
+{% switch enable_ipv6 %}
+{% case "true" %}
                                 "198.51.100.1",
                                 "fd01:d::c633:6401"
```
Comment (asridharan) [inline gen/dcos-config.yaml]: Just a minor comment:
Can't we do 
"198.51.100.1",
{%switch enable_ipv6%}
{%case false%}
{%case true%}
"fd01:d::c633:6401"
{%endswitch%}

Comment (urbanserj) [reply gen/dcos-config.yaml]: See the comma. I tried to do like below, but it looks ugly for me.

```
"198.51.100.1"
{%switch enable_ipv6%}
{%case false%}
{%case true%}
, "fd01:d::c633:6401"
{%endswitch%}
```

**Comment by asridharan** [comment]
@artemharutyunyan need approval for this to be merged so that DC/OS 1.11 can be installed in platforms with IPv6 disabled in the linux kernel as well.

File: gen/dcos-config.yaml (hunk 5 above)
Comment (asridharan) [reply gen/dcos-config.yaml]: 👍

**Comment by urbanserj** [comment]
@orsenthil this PR was approved. Please see the [comment](https://github.com/mesosphere/dcos-enterprise/pull/2354). Could you please remove `holding` label?

**Comment by asridharan** [comment]
@orsenthil we need this to be merged for 1.11. I think @artemharutyunyan gave the approval on this a few days back?

**Pull request merged**